integration-tests/contract-security-tickets-duplication.spec.ts

```diff
@@ -0,0 +1,156 @@
+import { CONFIGS } from './config';
+
+// TC-T-022: Duplicate ticket - duplicate transaction operation
+// TC-T-023: Duplicate ticket - Duplicate ticket - duplicate map containing tickets
+// TC-T-024: Duplicate ticket - duplicate big_map containing tickets
+
+CONFIGS().forEach(({ lib, rpc, setup }) => {
+  const Tezos = lib;
+
+  describe(`Test contracts using: ${rpc}`, () => {
+    beforeEach(async (done) => {
+      await setup();
+      done();
+    });
+
+    it('Verify creating ticket is not possible with duplicate transaction operation', async (done) => {
+      try {
+        const opJoin = await Tezos.contract.originate({
+          code: `   {  parameter (ticket string);
+      storage (option (ticket string));
+      code
+        {
+          UNPAIR; 
+          SWAP;
+          IF_NONE {
+                    SOME;
+                  }
+                  {
+                    PAIR;
+                    JOIN_TICKETS;
+                  };
+          NIL operation;
+          PAIR;
+        };      
+      }`,
+          init: 'None',
+        });
+
+        await opJoin.confirmation();
+        expect(opJoin.hash).toBeDefined();
+        expect(opJoin.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+        const opJoinContract = await opJoin.contract();
+        //expect(await opJoinContract.storage()).toBeTruthy();
+
+        const opDupOp = await Tezos.contract.originate({
+          code: `   {  parameter (ticket string);
+      storage (option (ticket string));
+      code
+        {
+          UNPAIR; 
+          SWAP;
+          IF_NONE {
+                    SOME;
+                  }
+                  {
+                    PAIR;
+                    JOIN_TICKETS;
+                  };
+          NIL operation;
+          PAIR;
+        };      
+      }`,
+          init: 'Unit',
+        });
+
+        await opDupOp.confirmation();
+        expect(opDupOp.hash).toBeDefined();
+        expect(opDupOp.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+        const opDupOpContract = await opDupOp.contract();
+        expect(await opDupOpContract.storage()).toBeTruthy();
+
+        await Tezos.contract
+          .at(opDupOpContract.address)
+          .then((contract) => {
+            return contract.methods.default(`${opJoinContract.address}`).send();
```

Comment:
The parameter of opDupOpContract.default should be a ticket but it is called with a contract address, is it expected?

Reply:
in the test, the CODE for originating opDupOpContrac was the wrong Michelson code.  Now it is corrected the test returns the expected exception "internal_operation_replay"